tests/unit/lms/services/assignment_test.py

```diff
@@ -1,188 +1,604 @@
-import uuid
 from unittest.mock import sentinel
 
 import pytest
+from sqlalchemy.orm.exc import MultipleResultsFound
 
-from lms.models import Assignment
 from lms.services.assignment import AssignmentService, factory
 from tests import factories
 
+MATCHING_TOOL_CONSUMER_INSTANCE_GUID = "matching_tool_consumer_instance_guid"
+MATCHING_RESOURCE_LINK_ID = "matching_resource_link_id"
+MATCHING_EXT_LTI_ASSIGNMENT_ID = "matching_ext_lti_assignment_id"
 
-class TestAssignmentService:
-    def test_get_without_ids(self, svc, assignment):
+
+class TestGet:
+    def test_at_least_one_of_resource_link_id_or_ext_lti_assignment_id_is_required(
+        self, svc
+    ):
         with pytest.raises(ValueError):
-            assignment = svc.get(
-                assignment.tool_consumer_instance_guid,
+            svc.get(
+                sentinel.tool_consumer_instance_guid,
                 resource_link_id=None,
                 ext_lti_assignment_id=None,
             )
 
-    def test_get_returns_None_if_theres_no_matching_assignment(self, svc):
-        assert not svc.get("TOOL_CONSUMER_INSTANCE_GUID", "RESOURCE_LINK_ID")
+    def test_it_raises_if_there_are_multiple_matching_assignments(self, svc):
+        # Create two assignments that will both match the query.
+        factories.Assignment(
+            tool_consumer_instance_guid=MATCHING_TOOL_CONSUMER_INSTANCE_GUID,
+            resource_link_id=MATCHING_RESOURCE_LINK_ID,
+            ext_lti_assignment_id=None,
+        )
+        factories.Assignment(
+            tool_consumer_instance_guid=MATCHING_TOOL_CONSUMER_INSTANCE_GUID,
+            resource_link_id=None,
+            ext_lti_assignment_id=MATCHING_EXT_LTI_ASSIGNMENT_ID,
+        )
+
+        with pytest.raises(MultipleResultsFound):
+            svc.get(
+                MATCHING_TOOL_CONSUMER_INSTANCE_GUID,
+                MATCHING_RESOURCE_LINK_ID,
+                MATCHING_EXT_LTI_ASSIGNMENT_ID,
+            )
 
-    def test_get_by_resource_link_id_only(self, svc, assignment):
-        retrieved_assignment = svc.get(
-            assignment.tool_consumer_instance_guid, assignment.resource_link_id
+    # If there's an assignment in the DB that has both the matching
+    # resource_link_id and the matching ext_lti_assignment_id then get() will
+    # find that assignment if called with:
+    #
+    # 1. The matching resource_link_id and no ext_lti_assignment_id
+    # 2. The matching ext_lti_assignment_id and no resource_link_id
+    # 3. Both the matching resource_link_id and the matching ext_lti_assignment_id.
+    @pytest.mark.parametrize(
+        "resource_link_id,ext_lti_assignment_id",
+        [
+            pytest.param(
```

Comment:
Never seen `pytest.param` before. Here it will use the id as the test name instead of just the combination of values? :+1:

Reply:
Yeah it goes into the test name in pytest's output. And also just helps to document each case for anyone reading the code (similar to a code comment)